extensions/amp-carousel/0.1/base-slides.js

```diff
@@ -196,4 +208,20 @@ export class BaseSlides extends BaseCarousel {
       this.autoplayTimeoutId_ = null;
     }
   }
+
+  /**
+  * Remove autoplay.
+  * @protected
+  */
+  removeAutoplay() {
+    this.clearAutoplay();
+    this.hasAutoplay_ = false;
+    this.shouldAutoplay_ = this.hasAutoplay_ && this.isLoopingEligible();
+    if (this.element.hasAttribute('autoplay')) {
```

Comment:
just the attribute check ignores the `isLoopingEligible` part. The value of `this.shouldAutoplay_` before it changes should be the condition.

Reply:
Agree with you there. Since we set `this.hasAutoplay_` to `false` we are guaranteed that `this.shouldAutoplay_` will be false. I can still guard it?

Reply:
I guess we don't even get here if it was not eligible for autoplay. I think the check to remove `loop` only if it was added because of autoplay is still wrong. Maybe it should be 

```
// loop added due to autoplay if element originally did not have `loop` attribute, so remove it.
if(!this.element.hasAttribute('loop') {
...
}
```

Reply:
looking at this again, I realize with don't actually have a signal to know whether `loop` was added because of autoplay or it was there from the beginning. We need a new variable to track that when autoplay adds `loop`.

Reply:
We can also just maintain state to see if `loop` was added by `autoplay` status addition

Reply:
Done in latest Patch - https://github.com/ampproject/amphtml/pull/18981/commits/1b9d55308c9a760006763e9f7e47169d3b2c48dd